File: Third_Year/DS/Practicas/P1/Método Factoria/PistaBaloncesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package DesignPatterns_FactoryMethod;


public class PistaBaloncesto extends Pista {
    private float alturaCanastas;

    public PistaBaloncesto (float tamanio,float alt){
        super(tamanio);
        this.alturaCanastas = alt;
        this.tipo = "Baloncesto";
    }

}

File: Third_Year/DS/Practicas/P1/Prototipo/PistaBaloncesto.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Prototipo

class PistaBaloncesto < Pista

  public
    def initialize(id)
      super(id)
      @players = Array.new
      @tipo = "Baloncesto"
    end

    def clonar
      return PistaBaloncesto.new(@id)
    end

    def addJugadores(jug)
      @players << jug
    end

    attr_accessor :players
end

end